code/modules/mentorhelp/mentor.dm

```diff
@@ -0,0 +1,71 @@
+GLOBAL_LIST_EMPTY(mentor_datums)
+GLOBAL_PROTECT(mentor_datums)
+
+GLOBAL_VAR_INIT(mentor_href_token, GenerateToken())
+GLOBAL_PROTECT(mentor_href_token)
+
+/datum/mentors
+	var/name = "someone's mentor datum"
+	var/client/owner // the actual mentor, client type
+	var/target // the mentor's ckey
+	var/href_token // href token for mentor commands, uses the same token used by admins.
+	var/mob/following
+
+/datum/mentors/New(ckey)
+	if(!ckey)
+		QDEL_IN(src, 0)
+		throw EXCEPTION("Mentor datum created without a ckey")
+		return
+	target = ckey(ckey)
+	name = "[ckey]'s mentor datum"
+	href_token = GenerateToken()
+	GLOB.mentor_datums[target] = src
+	//set the owner var and load commands
+	owner = GLOB.directory[ckey]
+	if(owner)
+		owner.mentor_datum = src
+		owner.add_mentor_verbs()
+		GLOB.mentors += owner
+
+/datum/mentors/proc/CheckMentorHREF(href, href_list)
+	var/auth = href_list["mentor_token"]
+	. = auth && (auth == href_token || auth == GLOB.mentor_href_token)
+	if(.)
+		return
+	var/msg = !auth ? "no" : "a bad"
+	message_admins("[key_name_admin(usr)] clicked an href with [msg] authorization key!")
+	if(CONFIG_GET(flag/debug_admin_hrefs))
+		message_admins("Debug mode enabled, call not blocked. Please ask your coders to review this round's logs.")
+		log_world("UAH: [href]")
+		return TRUE
+	log_admin_private("[key_name(usr)] clicked an href with [msg] authorization key! [href]")
+
+/proc/RawMentorHrefToken(forceGlobal = FALSE)
+	var/tok = GLOB.mentor_href_token
+	if(!forceGlobal && usr)
+		var/client/C = usr.client
+		to_chat(world, C)
+		to_chat(world, usr)
+		if(!C)
+			CRASH("No client for HrefToken()!")
+		var/datum/mentors/holder = C.mentor_datum
+		if(holder)
+			tok = holder.href_token
+	return tok
+
+/proc/MentorHrefToken(forceGlobal = FALSE)
+	return "mentor_token=[RawMentorHrefToken(forceGlobal)]"
+
+/proc/load_mentors()
+	GLOB.mentor_datums.Cut()
+	for(var/client/C in GLOB.mentors)
+		C.remove_mentor_verbs()
+		C.mentor_datum = null
+	GLOB.mentors.Cut()
+	var/list/lines = world.file2list("[global.config.directory]/mentors.txt")
```

Comment:
This is not good enough. This will be set to test merge again once database support has been added (I will likely be contributing to this PR myself)

Reply:
The database for mentors works but they dont get assigned as mentors in game, I dont know what the issue is

Reply:
database support completely fixed